Shared/PostList/PostListModel.swift

```diff
@@ -18,4 +18,107 @@ class PostListModel: ObservableObject {
+
+    func stripMarkdown(from string: String) -> String {
+        var strippedString = string
+        strippedString = stripHeadingOctothorpes(from: strippedString)
+        strippedString = stripImages(from: strippedString, keepAltText: true)
+        return strippedString
+    }
+
+    func stripHeadingOctothorpes(from string: String) -> String {
+        let newLines = CharacterSet.newlines
+        var processedComponents: [String] = []
+        let components = string.components(separatedBy: newLines)
+        for component in components {
+            if component.isEmpty {
+                continue
+            }
+            var newString = component
+            while newString.first == "#" {
+                newString.removeFirst()
+            }
+            if newString.hasPrefix(" ") {
+                newString.removeFirst()
+            }
+            processedComponents.append(newString)
+        }
+        let headinglessString = processedComponents.joined(separator: "\n\n")
+        return headinglessString
+    }
+
+    func stripImages(from string: String, keepAltText: Bool = false) -> String {
+        let pattern = #"!\[[\"]?(.*?)[\"|]?\]\(.*?\)"#
+        var processedComponents: [String] = []
+        let components = string.components(separatedBy: .newlines)
+        for component in components {
+            if component.isEmpty { continue }
+            var processedString: String = component
+            if keepAltText {
+                let regex = try? NSRegularExpression(pattern: pattern, options: [])
+                if let matches = regex?.matches(
+                    in: component, options: [], range: NSRange(location: 0, length: component.utf16.count)
+                ) {
+                    for match in matches {
+                        if let range = Range(match.range(at: 1), in: component) {
+                            processedString = "\(component[range])"
+                        }
+                    }
+                }
+            } else {
+                let range = component.startIndex..<component.endIndex
+                processedString = component.replacingOccurrences(
+                    of: pattern,
+                    with: "",
+                    options: .regularExpression,
+                    range: range
+                )
+            }
+            if processedString.isEmpty { continue }
+            processedComponents.append(processedString)
+        }
+        return processedComponents.joined(separator: "\n\n")
+    }
+
+    func extractLede(from string: String) -> String {
+        let truncatedString = string.prefix(80)
+        let terminatingPunctuation = ".。?"
+        let terminatingCharacters = CharacterSet(charactersIn: terminatingPunctuation).union(.newlines)
+
+        var lede: String = ""
+        let sentences = truncatedString.components(separatedBy: terminatingCharacters)
+        let firstSentence = sentences.filter { !$0.isEmpty }[0]
```

Comment:
Launching the app on a fresh simulator, when pressing the "new post" button, the app crashes with an `Index out of range` error on this line, specifically at the `[0]`. Full error:

```
Fatal error: Index out of range: file Swift/ContiguousArrayBuffer.swift, line 444
2020-11-10 12:54:35.924968-0500 WriteFreely-MultiPlatform[1422:26370] Fatal error: Index out of range: file Swift/ContiguousArrayBuffer.swift, line 444
```

Reply:
🤦 

Yeah, there's no guarantee of a first sentence — using `.first` instead of the `[0]` index will return an optional that we can check for. Fix coming in a second.